src/ripple/protocol/STBitString.h

```diff
@@ -25,7 +25,11 @@
 
 namespace ripple {
 
-template <std::size_t Bits>
+// The template parameter could be an unsigned type, however there's a bug in
+// gdb (last checked in gdb 12.1) that prevents gdb from finding the RTTI
+// information of a template parameterized by an unsigned type. This RTTI
+// information is needed to write gdb pretty printers.
+template <int Bits>
```

Comment:
There's a bit of an issue here since `Bits` is used as a parameter to the `base_uint` template which takes a `std::size_t` argument.  If someone passes a negative `Bits` then C++ conversion rules will turn that into a very big `std::size_t` for `base_uint`.  Yeah that will collapse early at run time. But I'd prefer to catch the problem at build time.

The simplest fix would be to add
```
static_assert(Bits > 0);
```
to the body of `STBitString`.  Probably best to place that `static_assert` before the `using value_type...`.

Reply:
Yep, good suggestion. Fixed in https://github.com/XRPLF/rippled/pull/4332/commits/ffef6fd66b4ebbd9e2f32d10b19c2b5218853626